File: 5 - Association Rule Learning/5a - Apriori/apriori.py
# Market Basket Analysis - Apriori

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Market_Basket.csv',header = None)

transactions = []
for i in range(dataset.shape[0]):
    transactions.append([str(dataset.values[i,j]) for j in range(dataset.shape[1])])


# Training Apriori on the dataset
from apyori import apriori
rules = apriori(transactions, 
                min_support = 0.004, 
                min_confidence = 0.2,
                min_lift = 3, 
                min_length = 2)

# Visualising the result
results = list(rules)
for i in range(10):
    print(results[i])

File: 5 - Association Rule Learning/5a - Apriori/apriori.R
# Market Basket Analysis - Apriori

# Importing the dataset
# install.packages('arules')
library(arules)
# dataset <- read.csv('Market_Basket.csv', header = FALSE)
dataset <- read.transactions('Market_Basket.csv', 
                             sep = ',', 
                             rm.duplicates = TRUE)
summary(dataset)
itemFrequencyPlot(dataset, topN = 50)    
     

# Training Apriori on the dataset
rules <- apriori(data = dataset,
                 parameter = list(support = 0.004, confidence = 0.2))

# Visualising the results
inspect(sort(rules, by = 'lift')[1:10])
